lib/AST/ASTPrinter.cpp

```diff
@@ -838,6 +843,11 @@ static bool hasNonMutatingSetter(const AbstractStorageDecl *ASD) {
   return setter && setter->isExplicitNonMutating();
 }
 
+static bool isSimpleStoredWithPublicSetter(const AbstractStorageDecl *asd) {
+  return asd->getImplInfo().isSimpleStored() &&
+    !asd->getAttrs().hasAttribute<SetterAccessAttr>();
```

Comment:
This doesn't seem right; it could be an `open` var with a `public(set)` (though I don't exactly know if we support that), or a `public` var with a `@usableFromInline` setter (though I don't know if we support *that),* or even a `public` var with a `public(set)` (which is only a warning).

Reply:
> it could be an `open` var with a `public(set)`

We do currently support this, so I'm gonna check for non-public-or-usableFromInline in the AccessLevel

> a `public` var with a `@usableFromInline` setter

Technically we support this on internal computed properties, but those won't get printed in interfaces at all.

> or even a `public` var with a `public(set)`

😢

Reply:
The check I'm doing now is if the setter's access level is less than the access level of the storage decl itself. Is that right?